PSAlphaFS.psm1

```diff
@@ -552,43 +576,101 @@ function Copy-LongItem
 		Else
 		{
 			$Destination_isDirectory = $true
-			if( -not ( $DirObject::Exists($Destination)))
-			{
-				$null = $DirObject::CreateDirectory($Destination)
-			} 
-			Else
+            #$DirObject::Exists seems to return false no matter what
+			#if( -not ( $DirObject::Exists($Destination)))
+			#{
+			#	$null = $DirObject::CreateDirectory($Destination)
+			#} 
+			#Else
+            #{
+			#	$Destination = $PathFSObject::Combine($Destination, $basename) 
+			#}
+            try {
+                $null = $DirObject::CreateDirectory($Destination)
+            }
+            catch [Alphaleonis.Win32.Filesystem.AlreadyExistsException]
 			{
-				$Destination = $PathFSObject::Combine($Destination, $basename)
+				
 			}
 		}#destination is a folder
         
 
-                
-		if($PathObject.EntryInfo.IsDirectory)
-		{
-			$fsObject = $DirObject
+		# Perform the copy     
+		
+		Write-Verbose -Message ("Copy-LongItem:`tCopying '{0}' to '{1}'" -f $Path, $Destination)
+        if($PathObject.EntryInfo.IsDirectory)
+		{			    
+            if ($isFile) {
+                Write-Warning -Message ("Copy-LongItem:`tCan not copy a directory '{0}' into a file {1}." -f $Path, $Destination)
+                return
+            }
+            else {
+                # Directory -> Directory
+                try {
+                    $DirObject::Copy($Path, $Destination, $copyOptions)  
+                }
+                catch [Alphaleonis.Win32.Filesystem.AlreadyExistsException] {
+                    # can get clever in here later and make it a mini robocop
+                    Write-Warning -Message ("Copy-LongItem:`tDirectory to Directory Copy with existing file conflicts not supported")                
```

Comment:
can we consider using a more direct warning message: 

     ("Copy-LongItem:`tDestination already exists [{0}]" -f $Destination)